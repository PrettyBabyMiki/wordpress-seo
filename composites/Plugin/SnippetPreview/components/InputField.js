import PropTypes from "prop-types";
import React from "react";
import { EditorState, Editor, Modifier } from "draft-js";
import colors from "../../../../style-guide/colors.json";
import styled from "styled-components";

const InputFieldContainer = styled.div`
	cursor: text;
	height: 16px;
	border: 1px solid ${ colors.$color_grey };
	width: 100%;
	padding: 3px 5px;
	margin: 5px 1px 1px 1px;
	background-color: ${ colors.$color_white };
	overflow: hidden;
	font-size: 14px;
	font-style: inherit;
	.public-DraftStyleDefault-block {
        white-space: nowrap;
        word-wrap: normal;
    }
`;

export default class InputField extends React.Component {
	/**
	 * Constructs an input field
	 *
	 * @param {Object} props The props for this input field.
	 */
	constructor( props ) {
		super( props );
		this.state = { editorState: EditorState.createEmpty() };
	}

	/**
	 * Renders the input field.
	 *
	 * @returns {ReactElement} The rendered input field.
	 */
	render() {
		return (
			<InputFieldContainer onClick={ this.focus.bind( this ) }>
				<Editor
					editorState={ this.state.editorState }
					handlePastedText={ this.handlePastedText.bind( this ) }
					onChange={ this.onChange.bind( this ) }
					placeholder={ this.props.placeholder }
					ref={ editor => {
						this.editor = editor;
					} }
					handleReturn={ () => "handled" }
				/>
			</InputFieldContainer>
		);
	}

	/**
	 * Changes when the editor state changes.
	 *
	 * @param {Object} editorState The editor state for the DraftJS object.
	 *
	 * @returns {void}
	 */
	onChange( editorState ) {
		this.setState( { editorState } );
	}

	/**
	 * Focuses the editor.
	 *
	 * @returns {void}
	 */
	focus() {
		this.editor.focus();
	}

	/**
	 * Handles pasted text into the editor.
	 *
	 * @param {string} text The inserted text.
	 *
	 * @returns {string} A result string.
	 */
	handlePastedText( text ) {
		this.onChange( EditorState.push(
			this.state.editorState,
			Modifier.replaceText(
				this.state.editorState.getCurrentContent(),
				this.state.editorState.getSelection(),
				text.replace( /\n|\r/g, " " )
			)
		) );
		return "handled";
	}
}

InputField.propTypes = {
	placeholder: PropTypes.string,
};

InputField.defaultProps = {
	placeholder: "",
};
