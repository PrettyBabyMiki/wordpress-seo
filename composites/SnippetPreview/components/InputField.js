import React from "react";
import { EditorState, Modifier, Editor } from "draft-js";
import colors from "../../../style-guide/colors.json";
import styled from "styled-components";

const styles = {
	root: {
		fontFamily: "Arial, Helvetica, sans-serif",
		fontSize: 14,
	},
	editor: {
		cursor: "text",
		height: 16,
		border: "1px solid",
		borderColor: colors.$color_grey,
		width: "20%",
		padding: "3px 5px",
		margin: "5px 1px 1px 1px",
		backgroundColor: colors.$color_white,
		overflow: "hidden",
	},
};

const EditorContainer = styled.div`
    .public-DraftStyleDefault-block {
        white-space: nowrap;
        word-wrap: normal;
    }
`;

export default class InputField extends React.Component {
	constructor( props ) {
		super( props );
		this.state = { editorState: EditorState.createEmpty() };
	}

	render() {
		return (
			<div style={ styles.root }>
				<div style={ styles.editor } onClick={ this.focus.bind( this ) }>
					<EditorContainer>
						<Editor
							editorState={ this.state.editorState }
							onChange={ this.onChange.bind( this ) }
							placeholder={ this.props.placeholder }
							ref="editor"
						/>
					</EditorContainer>
				</div>
			</div>
		);
	}

	onChange( editorState ) {
		this.setState( { editorState } );
	}

	focus() {
		this.refs.editor.focus();
	}
}
