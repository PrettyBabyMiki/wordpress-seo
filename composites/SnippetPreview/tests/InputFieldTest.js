import React from "react";
import renderer from "react-test-renderer";
import InputField from "../components/InputField";

test( "the input field matches the snapshot", () => {
	const component = renderer.create(
		<InputField placeholder="placeholder"/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
