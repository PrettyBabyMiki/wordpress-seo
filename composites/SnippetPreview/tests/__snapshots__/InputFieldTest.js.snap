// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the logo matches the snapshot 1`] = `
<div
  style={
    Object {
      "fontFamily": "Arial, Helvetica, sans-serif",
      "fontSize": 14,
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "border": "1px solid",
        "borderColor": "#ddd",
        "cursor": "text",
        "height": 16,
        "margin": "5px 1px 1px 1px",
        "overflow": "hidden",
        "padding": "3px 5px",
        "width": "20%",
      }
    }
  >
    <div
      className="sc-bdVaJa hPAemK"
    >
      <div
        className="DraftEditor-root"
      >
        <div
          className="public-DraftEditorPlaceholder-root"
        >
          <div
            className="public-DraftEditorPlaceholder-inner"
            id="placeholder-c2do4"
          >
            placeholder
          </div>
        </div>
        <div
          className="DraftEditor-editorContainer"
        >
          <div
            aria-activedescendant={undefined}
            aria-autocomplete={undefined}
            aria-describedby="placeholder-c2do4"
            aria-expanded={undefined}
            aria-haspopup={undefined}
            aria-label={undefined}
            aria-owns={undefined}
            autoCapitalize={undefined}
            autoComplete={undefined}
            autoCorrect={undefined}
            className="notranslate public-DraftEditor-content"
            contentEditable={true}
            data-testid={undefined}
            onBeforeInput={[Function]}
            onBlur={[Function]}
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onCopy={[Function]}
            onCut={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            onMouseUp={[Function]}
            onPaste={[Function]}
            onSelect={[Function]}
            role="textbox"
            spellCheck={false}
            style={
              Object {
                "outline": "none",
                "whiteSpace": "pre-wrap",
                "wordWrap": "break-word",
              }
            }
            suppressContentEditableWarning={true}
            tabIndex={undefined}
          >
            <div
              data-contents="true"
            >
              <div
                className=""
                data-block={true}
                data-editor="c2do4"
                data-offset-key="7d1b5-0-0"
              >
                <div
                  className="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="7d1b5-0-0"
                >
                  <span
                    data-offset-key="7d1b5-0-0"
                    style={Object {}}
                  >
                    <br
                      data-text="true"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
